Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1/defaults.go

```diff
@@ -323,4 +324,70 @@ func defaultSecurityContextConstraints(scc *SecurityContextConstraints) {
 		scc.AllowEmptyDirVolumePlugin = new(bool)
 		*scc.AllowEmptyDirVolumePlugin = true
 	}
+
+	defaultSecurityContextConstraintsVolumes(scc)
+}
+
+// defaultSecurityContextConstraintsVolumes defaults the volume slice of the SCC.
+// The volumes slice will take precedence in settings if it is set.
```

Comment:
this is going to be more complicated than you'd like... 
* old go clients will drop the volumes slice (so it will be nil), and set the booleans
* old naive clients will just touch the fields they know about (the booleans), and leave the other fields alone (like a responsible client), so the booleans have to take precedence over the volumes slice
* new go clients will serialize from the volumes slice internally to a correct external volumes slice and correct external booleans, so yay!
* new naive clients will have to set both the volumes slice and the booleans (or *possibly* omit the booleans and really hope they're talking to a *new* v1 server, which is iffy, and only works if our booleans are pointers and we can tell they were omitted)